From 147e2c1760de29127a9740e339c0af4f958dc071 Mon Sep 17 00:00:00 2001
From: Naveen Mamindlapalli <naveen.mamindlapalli@xilinx.com>
Date: Tue, 19 Mar 2013 17:29:17 +0530
Subject: [PATCH 532/628] usb: xilinxps_udc: Added checking for full speed
 gadget.

git://github.com/Xilinx/linux-xlnx.git xilinx-v14.7
commit e5b9aae142b36d572c7f7ce338be45456b68cb5c

Added checking for full speed gadget. The controller supports
both high speed and full speed in device mode.

Signed-off-by: Naveen Mamindlapalli <naveenm@xilinx.com>
Signed-off-by: Liming Wang <liming.wang@windriver.com>
---
 drivers/usb/gadget/xilinx_usbps_udc.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/drivers/usb/gadget/xilinx_usbps_udc.c b/drivers/usb/gadget/xilinx_usbps_udc.c
index 0a5b3b4..b5d55aa 100644
--- a/drivers/usb/gadget/xilinx_usbps_udc.c
+++ b/drivers/usb/gadget/xilinx_usbps_udc.c
@@ -1627,7 +1627,7 @@ static int xusbps_start(struct usb_gadget_driver *driver,
 	if (!udc_controller)
 		return -ENODEV;
 
-	if (!driver || (driver->max_speed < USB_SPEED_HIGH)
+	if (!driver || (driver->max_speed < USB_SPEED_FULL)
 			|| !bind || !driver->disconnect || !driver->setup)
 		return -EINVAL;
 
-- 
1.7.5.4

